src/server/UnconnectedMessageHandler.php

```diff
@@ -81,8 +82,13 @@ private function handle(OfflineMessage $packet, InternetAddress $address) : bool
 				$this->server->sendPacket(IncompatibleProtocolVersion::create($this->protocolAcceptor->getPrimaryVersion(), $this->server->getID()), $address);
 				$this->server->getLogger()->notice("Refused connection from $address due to incompatible RakNet protocol version (version $packet->protocol)");
 			}else{
+				$serverHasSecurity = false; // then relocate it ve make it available
+				$cookie = 0;
+				if ($serverHasSecurity) {
+					$cookie = crc32(Binary::writeLInt(mt_rand(0, 0xFFFFFFFF)) . Binary::writeLShort($address->getPort()) . $address->getIp());
```

Comment:
May be better if we moove all this encoding process in a Cookie class ? This can be the place for encoding / decoding stuff and passed through the parameters and property for better handling ?

Reply:
Is it right to open a class for a simple field?

Reply:
There is nothing big about creating a class, except the fact that it's annoying to copy the license header boilerplate.